ports/cheriot-rtos/machine_spi.c

```diff
@@ -0,0 +1,300 @@
+#include <stdio.h>
+
+#include "extmod/modmachine.h"
+#include "py/obj.h"
+#include "py/runtime.h"
+
+#define STRINGIFY(x) #x
+#define EX_STRINGIFY(x) STRINGIFY(x)
+
+#define SPI_MAX_BAUDRATE (25000000)
+#define SPI_MIN_BAUDRATE (SPI_MAX_BAUDRATE >> 16)
+#define DEFAULT_SPI_BAUDRATE (1000000)
+#define DEFAULT_SPI_POLARITY (0)
+#define DEFAULT_SPI_PHASE (0)
+#define DEFAULT_SPI_BITS (8)
+#define DEFAULT_SPI_FIRSTBIT (SPI_MSB_FIRST)
+
+typedef struct _spi_block_t {
+    uint32_t interruptState;
+    uint32_t interruptEnable;
+    uint32_t interruptTest;
+    uint32_t configuration;
+    uint32_t control;
+    uint32_t status;
+    uint32_t start;
+    uint32_t receiveFifo;
+    uint32_t transmitFifo;
+} spi_block_t;
```

Comment:
According to the style guide, "use underscore_case, not camelCase for all names."